### ISSUE TITLE
\genfrac fails with some delimiters (e.g., <>)

### ISSUE THREAD

**Issue by PeterLuschny** [mathjax/MathJax#1033]
Hi!

    \begin{equation}
    \sum_{k=0}^n (-1)^k \genfrac < > {0pt}{}{n}{k} = \frac{4^{n+1}-2^{n+1}}{n+1} B_{n+1},
    \end{equation}

worked until yesterday, today we have

    <math xmlns="http://www.w3.org/1998/Math/MathML" display="block">
      <merror>
        <mtext>\begin{equation}
    \sum_{k=0}^n&#xA0;(-1)^k&#xA0;\genfrac&#xA0;&lt;&#xA0;&gt;&#xA0;{0pt}{}{n}     {k}&#xA0;=&#xA0;\frac{4^{n+1}-2^{n+1}}{n+1}&#xA0;B_{n+1},
    \end{equation}</mtext>
      </merror>
    </math>

Regards
Peter Luschny

**Comment by CharlesAnifowose** [comment]
Are you using the CDN `//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML` ?

**Comment by PeterLuschny** [comment]
I was using "http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML".

I switched now to "2.4-latest". The full configuration can be seen on
http://luschny.de/math/zeta/The-Bernoulli-Manifesto.html

**Comment by CharlesAnifowose** [comment]
We've pulled back to `2.2-latest` instead of `latest` and we'll be monitoring to see if the issue persists.

Unfortunately, when MathJax fails like this, it does not seem to provide any mechanism for throwing an error, so we have to rely on people using the site telling us. Not ideal.

**Comment by dpvc** [comment]
@CharlesAnifowose, do you mean `2.4-latest` rather than `2.2-latest`?  I'm not sure the reason to go back that many versions.  Also, can you try `2.5-latest` and see if that helps?  If it is, indeed, a caching issue, that would resolve it while still giving you the improvements from 2.5.

@PeterLuschny, I have figured out the issue in your case, and it is a bug in v2.5.  There were changes to `\genfrac` to get the sizing of the delimiters right, and the conversion of the delimiters to their unicode values was being done too early for the new arrangement.  Here is a temporary fix for now until we push an update.  Add the following

``` html
<script type="text/x-mathjax-config">
if (MathJax.version === "2.5.0") {
  MathJax.Hub.Register.StartupHook("TeX AMSmath Ready",function () {
    var TEX = MathJax.InputJax.TeX,
        TEXDEF = TEX.Definitions;
    MathJax.InputJax.TeX.Parse.Augment({
      GetDelimiterArg: function (name) {
        var c = this.trimSpaces(this.GetArgument(name));
        if (c == "") return null;
        if (TEXDEF.delimiter[c]) return c;
        TEX.Error(["MissingOrUnrecognizedDelim","Missing or unrecognized delimiter for %1",name]);
      }
    });
  });
}
</script>
```

just before the script that loads `MathJax.js` itself.

**Comment by dpvc** [comment]
@CharlesAnifowose, there are a number of ways to get information about errors from MathJax.  For example, if you want to find out about TeX parsing errors, you can use

      MathJax.Hub.Register.MessageHook("TeX Jax - parse error",function (message) {
        // do something with the error.  message[1] will contain the data about the error.
      });

If you want to find out about "Math Processing Error" messages, use

      MathJax.Hub.Register.MessageHook("Math Processing Error",function (message) {
        //  do something with the error.  message[2] is the Error object that records the problem.
      });

There are similar messages for other input formats, and some messages about output like use of characters outside the fonts in use.  If you are interested in collecting such data, we have a [performance analysis tool](https://github.com/mathjax/MathJax-profiler) that can be used to collect lots of data about how MathJax is running.  If you want to discuss this further, the [MathJax-dev forum](https://groups.google.com/forum/?fromgroups#!forum/mathjax-dev) is probably the place for it.

**Comment by CharlesAnifowose** [comment]
@dpvc Thanks for the tip. That is quite useful. How do we add that to the docs for the next person who is looking for it?

**Comment by dpvc** [comment]
Well, making an issue or a pull request at https://github.com/mathjax/MathJax-docs/ would probably be the best way.

**Comment by CharlesAnifowose** [comment]
Thanks, I've gone ahead and created a pull request as well.

https://github.com/mathjax/MathJax-docs/issues/90

**Comment by dpvc** [comment]
==> Merged (into hotfix-2.5)
